### PR TITLE
Add codegen tests for name clashes

### PR DESCRIPTION
**Pull request by paulbalaji**
#### Description

Replace `illegal.schema` with unit tests using equivalent bundles.

#### Tests

Ran codegen tests, they pass.

Opened `test-project`, no longer logs name clashing errors.

#### Documentation

- [x] internal changelog

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Tests/Model/NameClashingTests.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+using System.IO;
+using System.Reflection;
+using Improbable.Gdk.CodeGeneration.Model;
+using Improbable.Gdk.CodeGeneration.Model.Details;
+using NUnit.Framework;
+
+namespace Improbable.Gdk.CodeGeneration.Tests.Model
+{
+    [TestFixture]
+    public class NameClashingTests
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Tests/Model/NameClashingTests.cs:11]: Can we also assert that we are logging errors/warnings (whichever it is) to the user?

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Tests/Model/NameClashingTests.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    {
+        [Test]
+        public void Clashing_command_in_component()
+        {
+            var bundleName = "clash_command_in_component";
+            var qualifiedName = "improbable.gdk.test.IllegalComponent";
+
+            var store = GetDetailsFromBundle(bundleName);
+
+            Assert.AreEqual(store.Components.Count, 1);
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Tests/Model/NameClashingTests.cs:21]: What's this assert for?

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Tests/Model/NameClashingTests.cs (hunk 1 above)
Comment (paulbalaji) [reply workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Tests/Model/NameClashingTests.cs:11]: I think there's a way, I'll try it out

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Tests/Model/NameClashingTests.cs (hunk 2 above)
Comment (paulbalaji) [reply workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Tests/Model/NameClashingTests.cs:21]: It's... not needed anymore

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Tests/Model/NameClashingTests.cs (hunk 1 above)
Comment (paulbalaji) [reply workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Tests/Model/NameClashingTests.cs:15]: ye it works

**Pull request merged**